addons/l10n_ro_edi_stock/models/stock_picking.py

```diff
@@ -0,0 +1,864 @@
+    l10n_ro_edi_stock_enable_fetch = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_fetch')
+    l10n_ro_edi_stock_enable_send_info = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_send_info')
+    l10n_ro_edi_stock_enable_amend = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_amend')
+
+    l10n_ro_edi_stock_fields_readonly = fields.Boolean(compute='_compute_l10n_ro_edi_stock_fields_readonly')
+
+    ################################################################################
+    # Reset Data Methods
+    ################################################################################
+
+    @api.onchange('l10n_ro_edi_stock_operation_type')
+    def _l10n_ro_edi_stock_reset_location_type(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_start_loc_type = 'location'
+            picking.l10n_ro_edi_stock_end_loc_type = 'location'
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_start_loc_type')
+    def _l10n_ro_edi_stock_reset_start_location_data(self):
+        for picking in self:
+            picking._l10n_ro_edi_stock_reset_location_data('start')
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_end_loc_type')
+    def _l10n_ro_edi_stock_reset_end_location_data(self):
+        for picking in self:
+            picking._l10n_ro_edi_stock_reset_location_data('end')
+
+    def _l10n_ro_edi_stock_reset_location_data(self, location: str):
+        for picking in self:
+            picking[f'l10n_ro_edi_stock_{location}_bcp'] = False
+            picking[f'l10n_ro_edi_stock_{location}_customs_office'] = False
+            picking[f'l10n_ro_edi_stock_{location}_state_id'] = False
+            picking[f'l10n_ro_edi_stock_{location}_zip'] = False
+            picking[f'l10n_ro_edi_stock_{location}_city'] = False
+            picking[f'l10n_ro_edi_stock_{location}_street'] = False
+            picking[f'l10n_ro_edi_stock_{location}_other_info'] = False
+
+    ################################################################################
+    # Compute Methods
+    ################################################################################
+
+    @api.depends('l10n_ro_edi_stock_operation_type')
+    def _compute_l10n_ro_edi_stock_available_operation_scopes(self):
+        for picking in self:
+            allowed_scopes = [c for c, dummy in OPERATION_SCOPES]
+
+            if picking.l10n_ro_edi_stock_operation_type:
+                allowed_scopes = OPERATION_TYPE_TO_ALLOWED_SCOPE_CODES.get(picking.l10n_ro_edi_stock_operation_type, ("9999",))
+
+                if picking.l10n_ro_edi_stock_operation_scope not in allowed_scopes:
+                    picking.l10n_ro_edi_stock_operation_scope = None
+            elif picking.l10n_ro_edi_stock_operation_scope:
+                picking.l10n_ro_edi_stock_operation_scope = None
+
+            picking.l10n_ro_edi_stock_available_operation_scopes = ','.join(allowed_scopes)
+
```

Comment:
There is a problem here. You must only and always assign `l10n_ro_edi_stock_available_operation_scopes`.
Why assigning `l10n_ro_edi_stock_operation_scope` ?
Why `allowed_scopes` is overridden with something else than the original data structure?

```py
    l10n_ro_edi_stock_available_operation_scopes = fields.Char(compute='_compute_l10n_ro_edi_stock_available_operation_scopes')

```